Userland/Libraries/LibC/stdlib.cpp

```diff
@@ -719,8 +742,36 @@ char* ptsname(int fd)
 
 int ptsname_r(int fd, char* buffer, size_t size)
 {
-    int rc = syscall(SC_ptsname, fd, buffer, size);
-    __RETURN_WITH_ERRNO(rc, rc, -1);
+    struct stat stat;
+    if (fstat(fd, &stat) < 0)
+        return -1;
+
+    // Note: glibc says that for a maximum size for `int' of 8 bytes we never need more than 20 digits.
+    // we create buffer to use to print the number in and it's 21 bytes to accomodate a null byte at the end.
+    char number_converted_string[21];
+    // Clean the array, just in case...
+    memset(&number_converted_string, 0, 21);
+    const char* devpts = "/dev/pts/";
+    const size_t devpts_length = strlen(devpts);
```

Comment:
I think it would be cleaner to just use a `StringBuilder` and here, and just copy that to the destination buffer.

Reply:
I thought so too, but what about ports that compile without C++ support?

Reply:
So after you explained we link LibC against libstdc++ and therefore we can use C++ code, this seems like a good idea, and I will do that :)